app/src/views/private/components/export-sidebar-detail.vue

```diff
@@ -114,7 +114,14 @@
 
 				<div class="field half-right">
 					<p class="type-label">{{ t('limit') }}</p>
-					<v-input v-model="exportSettings.limit" type="number" :min="-1" :step="1" :placeholder="t('unlimited')" />
+					<v-input
+						v-model="exportSettings.limit"
+						type="number"
+						:min="queryLimitMax === Infinity ? -1 : 0"
+						:max="queryLimitMax"
+						:step="1"
+						:placeholder="queryLimitMax === Infinity && t('unlimited')"
```

Comment:
Noticed a warning being thrown for the `placeholder` whenever the limit is changed.
 
<img width="684" alt="image" src="https://github.com/directus/directus/assets/26413686/dde01d22-35d0-4977-93fc-7bbb69990dd9">